### ISSUE TITLE
Main navigation menu not fully operable with a keyboard

### ISSUE THREAD

**Issue by afercia** [WordPress/twentynineteen#266]
In Twenty Seventeen, it is possible to navigate the main menu with a keyboard tabbing in both directions. Pressing `tab` navigates through all the menu items, pressing `shift+tab` navigates through all the menu items in reverse order.

As far as I see, in Twenty Nineteen navigating forwards allows to go through all the menu items, including all the sub-items.

Instead, when navigating backwards, the sub-items are skipped.

**Comment by westonruter** [comment]
Which browser(s) was this observed in?

**Comment by afercia** [comment]
All of them. Really not a browsers issue.

**Comment by westonruter** [comment]
I think I understand why this is happening: when tabbing forward you start with focus on the root link of the nav submenu. At this point, the links below are displayed and so they can then be tabbed to. However, if you tab past the last link in a submenu and then tab backwards, the last link in the submenu is no longer displayed and so it cannot be focused, meaning the keyboard jumps back to the previous closest focusable element: the root nav menu item.

**Comment by afercia** [comment]
Yep. In CSS there's no selector to target an ancestor. The relatively new `focus-within` could be used to build a pure-CSS solution, see for example https://www.scottohara.me/blog/2017/05/14/focus-within.html. Unfortunately, `focus-within` is not supported by Edge and IE. I've heard there are polyfills but I've never played with them so I can't say if they're good. If they're good enough, it would be great to explore a pure-CSS solution.
Alternatively, Twenty Seventeen solves this issue with a little bit of JavaScript, I'd suggest to have a look at it.

**Comment by afercia** [comment]
The ones I've found are:
https://github.com/jonathantneal/focus-within
https://github.com/jonathantneal/postcss-focus-within

**Comment by westonruter** [comment]
I just found that the issue is not due to `:focus-within`. Rather, it's because the submenus get `display:none` when not displayed:

https://github.com/WordPress/twentynineteen/blob/3aa5fbee914e69002540fca20c160d56290ad509/sass/navigation/_menus.scss#L73

In Twenty Seventeen, however, these submeus get `display:block`:

https://github.com/WordPress/wordpress-develop/blob/5d92e0c0a950399c53294a41000305ddcf47efb5/src/wp-content/themes/twentyseventeen/style.css#L3515-L3519

For both Twenty Seventeen and Twenty Nineteen, the submenu that is not shown gets positioned off screen:

* https://github.com/WordPress/wordpress-develop/blob/5d92e0c0a950399c53294a41000305ddcf47efb5/src/wp-content/themes/twentyseventeen/style.css#L3545-L3553
* https://github.com/WordPress/twentynineteen/blob/3aa5fbee914e69002540fca20c160d56290ad509/sass/navigation/_menus.scss#L76-L79

So I think the only fix needed is to remove the `display:none`. This would likely depend on a the no-JS strategy for showing submenus in #280/#337.

**Comment by allancole** [comment]
This should all be fixed in #337 and I’m going to add a fallbacks for `:focus-within` to the build tool that will hopefully do the trick. I ended up using some Javascript to correct the menu behavior on touch-devices and had to bring back the `aria-expanded="false"` so the state should change properly now. 

It definitely needs some more testing so if anything is off, just let me know.

**Comment by richsalvucci** [comment]
I tested on a Mac with Chrome, Safari and Fire Fox and the menu is working as expected with the keyboard functionality using the tab key.

**Comment by c4rl0sbr4v0** [comment]
Hi!
Tested on a Max with Firefox, I tried to go to a third level with the keyboard using right arrow and couldn't, should we do that?
